Error trying to build unit tests on hobart

Pointing to my branch from #1343 , I tried running this on hobart:

```
tools/unit_testing/run_tests.py --build-dir `mktemp -d --tmpdir=. unit_tests.XXXXXXXX`
```

and got this error message:

```
CMake Error at unit_tests.51Nwz7QH/__command_line_test__/__command_line_test__/Macros.cmake:189 (execute_process):
  Syntax error in cmake code at

    /home/sacks/cime/unit_tests.51Nwz7QH/__command_line_test__/__command_line_test__/Macros.cmake:189

  when parsing string

    ${NETCDF_PATH

  syntax error, unexpected $end, expecting } (13)
Call Stack (most recent call first):
  src/externals/CMake/CIME_utils.cmake:33 (include)
  CMakeLists.txt:12 (include)
```

This is pointing to this line in Macros.cmake:

```
execute_process(COMMAND ${NETCDF_PATH OUTPUT_VARIABLE CIME_TEMP_SHELL0 OUTPUT_STRIP_TRAILING_WHITESPACE)
```

which I see is missing an ending `}`

--- a/config/cesm/machines/config_compilers.xml
+++ b/config/cesm/machines/config_compilers.xml
@@ -667,7 +667,7 @@ using a fortran linker.
   <MPI_LIB_NAME MPILIB="mvapich2"> mpich</MPI_LIB_NAME>
   <NETCDF_PATH>$ENV{NETCDF_PATH}</NETCDF_PATH>
   <SLIBS>
-    <append>$SHELL{${NETCDF_PATH}/bin/nf-config --flibs}</append>
+    <append>-L$NETCDF_PATH/lib -lnetcdff -lnetcdf</append>
   </SLIBS>
 </compiler>
 

